features/FEATURE_BLE/ble/Gap.h

```diff
@@ -321,6 +269,617 @@ class GapAdvertisingData;
+        ) { }
+
+        virtual void onDisconnection(const ble::DisconnectionEvent &event) { }
+
+        /**
+         * Function invoked when the current transmitter and receiver PHY have
+         * been read for a given connection.
+         *
+         * @param status Status of the operation: BLE_ERROR_NONE in case of
+         * success or an appropriate error code.
+         *
+         * @param connectionHandle: The handle of the connection for which the
+         * PHYs have been read.
+         *
+         * @param txPhy PHY used by the transmitter.
+         *
+         * @param rxPhy PHY used by the receiver.
+         */
+        virtual void onReadPhy(
+            ble_error_t status,
+            ble::connection_handle_t connectionHandle,
+            ble::phy_t txPhy,
+            ble::phy_t rxPhy
+        ) { }
+
+        /**
+         * Function invoked when the update process of the PHY has been completed.
+         *
+         * The process can be initiated by a call to the function setPhy, the
+         * local bluetooth subsystem or the peer.
+         *
+         * @param status Status of the operation: BLE_ERROR_NONE in case of
+         * success or an appropriate error code.
+         *
+         * @param connectionHandle: The handle of the connection on which the
+         * operation was made.
+         *
+         * @param txPhy PHY used by the transmitter.
+         *
+         * @param rxPhy PHY used by the receiver.
+         *
+         * @note Success doesn't mean the PHY has been updated it means both
+         * ends have negotiated the best PHY according to their configuration and
+         * capabilities. The PHY currently used are present in the txPhy and
+         * rxPhy parameters.
+         */
+        virtual void onPhyUpdateComplete(
+            ble_error_t status,
+            ble::connection_handle_t connectionHandle,
+            ble::phy_t txPhy,
+            ble::phy_t rxPhy
+        ) {  }
+
+    protected:
+        /**
+         * Prevent polymorphic deletion and avoid unnecessary virtual destructor
+         * as the Gap class will never delete the instance it contains.
+         */
+        ~EventHandler() { }
+    };
+
+    /**
+     * Representation of a whitelist of addresses.
+     */
+    struct Whitelist_t {
+        /**
+         * Pointer to the array of the addresses composing the whitelist.
+         */
+        BLEProtocol::Address_t *addresses;
+
+        /**
+         * Number addresses in this whitelist.
+         */
+        uint8_t size;
+
+        /**
+         * Capacity of the array holding the addresses.
+         */
+        uint8_t capacity;
+    };
+
+
+    /** Check controller support for a specific feature.
+     *
+     * @param feature Feature to check.
+     * @return True if feature is supported.
+     */
+    virtual bool isFeatureSupported(ble::controller_supported_features_t feature)
+    {
+        // TODO: deal with legacy implementation
```

Comment:
I guess this should return `false` by default

Reply:
Fixed